### DISCUSSION TITLE
AzureDeveloperCliCredential for PowerShell

### DISCUSSION THREAD

**Discussion by peterbud** [Azure/azure-dev#3094] [Q&A]
In the python `azure.identity` package there is a `AzureDeveloperCliCredential()`  method. What is the designated equivalent of this method in PowerShell?

**Comment by vhvb1989** [comment]
@chlowell , do you know about credentials on PowerShell?  Is that under Azure SDK umbrella?

**Comment by chlowell** [comment]
The .NET equivalent is the [AzureDeveloperCliCredential](https://learn.microsoft.com/dotnet/api/azure.identity.azuredeveloperclicredential?view=azure-dotnet) class in [Azure.Identity](https://www.nuget.org/packages/Azure.Identity#readme-body-tab). (I'm assuming you want to acquire tokens from the Azure Developer CLI in a PowerShell script; please add more details if I've misunderstood.)

**Reply by peterbud** [reply]
Yes, I'd like to get tokens from the Azure Developer CLI in a PowerShell script.

I have tried the Azure.Identity package first. It is installed for CurrentUser scope, as confirmed by:
```powershell
Get-Package | Where-Object { $_.Name -like '*identity*' }

Name                           Version          Source                           ProviderName
----                           -------          ------                           ------------
Azure.Identity                 1.10.4           C:\Users\...\AppData\Loc... NuGet
```

However when I try using it:
```powershell
$credential = New-Object -TypeName Azure.Identity.AzureDeveloperCliCredential
```

I get an error: `Cannot find type [Azure.Identity.AzureDeveloperCliCredential]: verify that the assembly containing this type is loaded.`

So I have checked how the [python library](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/identity/azure-identity/azure/identity/_credentials/azd_cli.py#L27) is doing, and as I see it is using the `azd` CLI, and at the end I'm using that approach:

```powershell
command = "azd auth token --output json --scope https://graph.microsoft.com/.default"
$result = Invoke-Expression $command
$token = $result -join "" | ConvertFrom-Json | Select-Object -ExpandProperty token
```

**Comment by vhvb1989** [comment]
Here's a related conversation from the Teams channel: https://teams.microsoft.com/l/message/19:ug5DKpbpRJbBtgoSQAXdynHgthpALgePNJ8wb5AI8HM1@thread.tacv2/1700007689925?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=c99e8504-1250-4d95-a824-9778e0e626c7&parentMessageId=1700007689925&teamName=Azure%20Developer%20CLI%20(azd)%20Partners&channelName=General&createdTime=1700007689925

And related issues: https://github.com/Azure/azure-dev/issues/2975

You will find info about this example that uses `azd auth token` from a script: https://github.com/Azure-Samples/azure-openai-speech-to-speech-chat/blob/main/scripts/setkeys.sh

And also info about how you can make azd to use the `az` for auth, in case you might prefer to have one only source of auth with `az`

**Reply by peterbud** [reply]
It seems that the link:  https://github.com/Azure-Samples/azure-openai-speech-to-speech-chat/blob/main/scripts/setkeys.sh is not working?
